otherlibs/systhreads/st_stubs.c

```diff
@@ -360,7 +363,9 @@ static void caml_thread_remove_and_free(caml_thread_t th)
 }
 
 /* Reinitialize the thread machinery after a fork() (PR#4577) */
-/* TODO(engil): more work on the multicore fork machinery. */
+/* TODO(engil): more work on the multicore fork machinery.
+ * TODO(nick): in particular this only discards thread information for
+ * the current domain, not for all domains. */
```

Comment:
This looks ok to me; this is in fact all domains as fork raises an exception if more than one domain is running.